src/middlewared/middlewared/plugins/service_/services/mdns.py

```diff
@@ -13,15 +13,17 @@ class MDNSService(SimpleService):
     systemd_unit = "avahi-daemon"
 
     async def start(self):
+        failover_status = await self.middleware.call("failover.status")
         announce = (await self.middleware.call("network.configuration.config"))["service_announcement"]
-        if not announce["mdns"]:
+        if not announce["mdns"] or failover_status not in ['SINGLE', 'MASTER']:
```

Comment:
I'd rather stop the avahi service on a BACKUP event since that's the only sure way of knowing what state we're in.

Reply:
Okay. Sounds fair.